src/Open-SAMP-API/Client/SAMPFunctions.cpp

```diff
@@ -2,48 +2,56 @@
 #include "GTAFunctions.hpp"
 #include <Shared/PipeMessages.hpp>
 #include <boost/algorithm/string.hpp>
+#include <boost/scope_exit.hpp>
 
-int Client::SAMPFunctions::ReadGTACmdArgument(char *option, char *&str, int max_len) {
-	char *szCommandLine = new char[512];
+int Client::SAMPFunctions::ReadGTACmdArgument(const char *option, char *&str, const int max_len) {
+	auto *szCommandLine = new char[512];
 	ZeroMemory(szCommandLine, 512);
 
-	if (GTAFunctions::GetGTACommandLine(szCommandLine, 512) == 0)
+	if (GTAFunctions::GetGTACommandLine(szCommandLine, 512) == 0) {
+		delete[] szCommandLine;
 		return 0;
+	}
 
 	// The gta_sa.exe process of SAMP looks something like this:
 	// PATH/TO/gta_sa.exe -n NAME -h IP -p PORT
 	// In here, we attempt to read any attributes that were passed in there
-	char *context = NULL;
-	char *token = strtok_s(szCommandLine, " ", &context);
+	char *context = nullptr;
+	auto *token = strtok_s(szCommandLine, " ", &context);
 
-	while (token != NULL) {
-		token = strtok_s(NULL, " ", &context);
+	while (token != nullptr) {
+		token = strtok_s(nullptr, " ", &context);
 		if (strcmp(token, option) == 0)
 			break;
 	}
 
-	if (token != NULL)
+	if (token != nullptr)
 	{
-		token = strtok_s(NULL, " ", &context);
+		token = strtok_s(nullptr, " ", &context);
 		strcpy_s(str, max_len, token);
 	}
 
 	delete[] szCommandLine;
-	return token != NULL;
+	return token != nullptr;
 }
 
-EXPORT int Client::SAMPFunctions::GetServerIP(char *&ip, int max_len)
+EXPORT int Client::SAMPFunctions::GetServerIP(char *&ip, const int max_len)
 {
 	return ReadGTACmdArgument("-h", ip, max_len);
 }
 
 EXPORT int Client::SAMPFunctions::GetServerPort()
 {
-	char *portStr = new char[5];
-	ZeroMemory(portStr, 5);
-	if (ReadGTACmdArgument("-p", portStr, 5)) {
+	auto *portStr = new char[6];
+	ZeroMemory(portStr, 6);
+
+	BOOST_SCOPE_EXIT_ALL(portStr) {
+		delete[] portStr;
+	};
+
+	if (ReadGTACmdArgument("-p", portStr, 6)) {
```

Comment:
Didn't know whether we should use 5 or 6.
Guess with 6, we can still end up with a non-terminated string?

Reply:
Then again, we do the same thing down in `GetPlayerId` where we allocate a `char[25]` and call `GetPlayerName` with `max_len=25`.
There, too, we should never exceed the SAMP limit of 20 chars.